assets/js/modules/search-console/components/dashboard/DashboardPopularKeywordsWidget.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * DashboardPopularKeywordsWidget component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { numberFormat } from '../../../../util';
+import { getDataTableFromData, TableOverflowContainer } from '../../../../components/data-table';
+import whenActive from '../../../../util/when-active';
+import PreviewTable from '../../../../components/preview-table';
+import Layout from '../../../../components/layout/layout';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+const { useSelect } = Data;
+
+function DashboardPopularKeywordsWidget() {
+	const {
+		baseServiceURL,
+		data,
+		error,
+		loading,
+		searchConsolePropertyMainURL,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+		const domain = store.getPropertyID();
+		const args = {
+			dateRange: select( CORE_USER ).getDateRange(),
+			dimensions: 'query',
+			limit: 10,
+		};
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		if ( url ) {
+			args.url = url;
+		}
+
+		return {
+			data: store.getReport( args ),
+			error: store.getErrorForSelector( 'getReport', [ args ] ),
+			loading: store.isResolving( 'getReport', [ args ] ),
+			baseServiceURL: store.getServiceURL( {
+				path: '/performance/search-analytics',
+				query: {
+					resource_id: domain,
+					num_of_days: 28,
```

Comment:
There's a quick win we can get here, linking to the actually relevant result if in the single URL view: If there is a current entity (`url` is set), let's also pass another query argument here which needs to be like `page: !${ url }`. The `!` ensures it will be looking for an exact URL match, you can try it out by visiting a URL like this in your Search Console property.

The current behavior ignores the current URL, which isn't as great, so let's fix this in the new widget.

Reply:
Updated.